source/client.c

```diff
@@ -851,6 +851,19 @@ int aws_mqtt_client_connection_connect(
             (uint64_t)connection_options->ping_timeout_ms, AWS_TIMESTAMP_MILLIS, AWS_TIMESTAMP_NANOS, NULL);
     }
 
+    /* Keep alive time should always be greater than the timeouts. */
+    if (AWS_UNLIKELY(
```

Comment:
Couldn't this be a warning, and we just clamp the value for you?

Reply:
That's going to be a hidden behavior that we can't actually get right 100% of the time. We already have default values, if someone sets them, and they're wrong, I think we should let them know immediately.

Reply:
Ok, fair enough, sounds good.